### ISSUE TITLE
Angular2 has too many dependencies!

### ISSUE THREAD

**Issue by thany** [angular/angular#11410]
Angular 2.0 has way too many depedencies. It depends on...

* A shim for IE11 - IE11 is not old enough for it to need to depend on a shim.
* Zone,js - Throwing this out gives an error like `ng is not defined`. Is this where Angular lives or something??
* Reflect.js - Throwing this out gives another error message. Can you not include this in the library?
* RxJS - Some sort of reactive library. Isn't this what Angular is supposed to offer?

And after that, there are no less than five huge other libraries that seem to comprise some sort of baseline Angular2 library.

So as for the dependencies that should be included in the library core: 4 requests totalling 144KB. _That's nearly the size of Angular 1.x!_
Angular2 itself is even worse: 5 requests totalling 279KB.

So we're talking about a library that ballooned to almost triple the size of 423KB, and nine times the number of requests. **Just to get started.**

So basically what I'm asking is: why can't we have just one thing to include, preferably of a little bit more of a sane filesize?

**Comment by mattlewis92** [comment]
The reason for not having these dependencies bundled with angular is because they should eventually all become available natively in the browser e.g. you can see the specs here for [observables](https://github.com/tc39/proposal-observable), [zones](https://github.com/domenic/zones), [reflect-metadata](https://github.com/rbuckton/ReflectDecorators/blob/master/spec/metadata.md). They are all interchangeable with other libraries that match the spec hence why angular is unopinionated about forcing you to use a certain one e.g. you can use core-js for reflect metadata.

Out the box angular2 is quite large but there has been good progress on getting production builds smaller, by jumping through some hoops you can get a [hello world app to 50kb](http://blog.mgechev.com/2016/06/26/tree-shaking-angular2-production-build-rollup-javascript/)

Admittedly right now it's a PITA to follow and implement all of this, but eventually specs will be finished and browsers implement the polyfills angular requires, and the CLI will take care of all the production bundling. Or if you really don't like it use something else with a more stable ecosystem like react 😜

**Comment by vicb** [comment]
@mattlewis92 many thanks for this explanation. exactly that.

**Comment by thany** [comment]
The point is dat Angular 1.5 does not need any polyfills, and contains just everything it needs. Simple, effective, understandable, maintainable, slim, and sane. That, or it's so good that it doesn't need all that baggage in the first place.

With Angular 2, we suddenly need to have polyfills for features that no browser currently supports, and won't support for an undecidedly long time. It could be freaking YEARS. Or, can you tell me which browser supports any of these features? If "none", then it should just be included in the library. You can slim it down once browsers start to ship these features.

So in short: if a feature is support by NO browser, either don't use it or workaround it in the library. Don't pretend you don't need to worry about it by throwing in yet another polyfill.

Angular 1.x contains everything it needs. That is a GOOD thing. Slim it down when it doesn't anymore. Then everyone benefits, and noone falls into maintenance hell with a billion dependencies.

**Comment by qdouble** [comment]
@thany the solution for that would be for none of the community to use any modern web features. I doubt the advancement of a brand new framework should be all about keeping us compatible with ie8, no polyfills required.

**Comment by vicb** [comment]
@thany we think doing this will help moving the platform forward

**Comment by mattlewis92** [comment]
As I said in my original message, there are alternative implementations of all the polyfills. In the same way that angular leaves it up to you to choose your own es6 polyfill, the same is true for the others. 

Angular 1.x was written during a much slower and arguably simple time of JS development, angular 2 reflects today's state of modern JS. Eventually things will slow down and stabilise, it just won't happen overnight, but the way angular 2 is today is just preparing for that time.

**Comment by mohammedzamakhan** [comment]
> Out the box angular2 is quite large but there has been good progress on getting production builds smaller, by jumping through some hoops you can get a hello world app to 50kb

@mattlewis92 Will treeshaking really help for large scale application? when you have the whole app using almost all the features of angular 2, and then even treeshaking would give 200k size of angular 2 code.

I am an angular guy, but I keep performance in mind, with more than 400k of just framework stuff, it makes me feel more comfy with angular 1.x 😄

**Comment by qdouble** [comment]
@mohammedzamakhan if an app is of considerable size, you'll need to have lazy loading. No way to make a hundred features, modules and templates 50KB :)

**Comment by mattlewis92** [comment]
When your app gets too big, that's when you should start using code splitting and lazy loading of routes. The new router and webpack make this really easy. That way your initial entry bundle is quite small and everything else is lazy loaded. 

As time goes by the angular team will develop more and more techniques to make the production builds even smaller. But if you're happy with angular 1.x just stick with it, it's a solid and mature framework, angular 2 still has a way to go until it can compete in every way.

**Comment by thany** [comment]
@mattlewis92
> As I said in my original message, there are alternative implementations of all the polyfills. In the same way that angular leaves it up to you to choose your own es6 polyfill, the same is true for the others.

So let me rephrase my comment then: even if I want to support *only* the latest browsers, and not care about any IE, I should not need polyfills.

@vicb 
> we think doing this will help moving the platform forward

By providing polyfills only folder old browsers, totally agree. But adding an excess amount of baggage even for modern browsers, no. That's moving backward.
